autoscale-core/src/main/java/com/hpe/caf/autoscale/core/ScalerThread.java

```diff
@@ -153,11 +155,33 @@ private void handleAnalysis()
      * @param amount the requested number of instances to scale up by
      * @throws ScalerException if the scaling operation fails
      */
-    private void scaleUp(final int amount)
+    private void scaleUp(final int amount, final InstanceInfo info)
         throws ScalerException
     {
-        LOG.info("Triggering scale up of service {} by amount {}", serviceRef, amount);
+        LOG.info("Attempting scale up of service {} by amount {}", serviceRef, amount);
         scaler.scaleUp(serviceRef, amount);
+        try {
+            Thread.sleep(info.getMaxLaunchDelaySeconds());
+            InstanceInfo refreshedInsanceInfo = scaler.getInstanceInfo(serviceRef);
+            this.firstAttempt = System.currentTimeMillis();
+            Thread.sleep(refreshedInsanceInfo.getMaxLaunchDelaySeconds());
+            int instanceDifferences = refreshedInsanceInfo.getInstances() - refreshedInsanceInfo.getTotalRunningAndStageInstances();
+            while (refreshedInsanceInfo.getInstances() > refreshedInsanceInfo.getTotalRunningAndStageInstances()) {
+                Thread.sleep(instanceDifferences * 10 * 1000);
+                refreshedInsanceInfo = scaler.getInstanceInfo(serviceRef);
+                if (refreshedInsanceInfo.getInstances() > refreshedInsanceInfo.getTotalRunningAndStageInstances()) {
+                    if (!governor.makeRoom(serviceRef) || timelimitExceeded()) {
+                        throw new ScalerException(
+                            "Unable to scale service " + serviceRef + " due to an inability to make room for it on the orchestrator.");
+                    }
+                    instanceDifferences = refreshedInsanceInfo.getInstances() - refreshedInsanceInfo.getTotalRunningAndStageInstances();
+                }
+            }
+        } catch (final InterruptedException ex) {
+            Thread.interrupted();
+            throw new RuntimeException(ex);
```

Comment:
This doesn't seem right.  If the `InterruptedException` can't be handled then why isn't it being propagated?